tests/test_mapping.py

```diff
@@ -0,0 +1,98 @@
+# PyZX - Python library for quantum circuit rewriting 
+#        and optimization using the ZX-calculus
+# Copyright (C) 2018 - Aleks Kissinger and John van de Wetering
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#    http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import unittest
+import random
+import sys
+if __name__ == '__main__':
+    sys.path.append('..')
+    sys.path.append('.')
+
+try:
+    import numpy as np
+    from pyzx.tensor import tensorfy, compare_tensors
+except ImportError:
+    np = None
+
+try:
+    import quimb as qu
+    import quimb.tensor as qtn
+except ImportError:
+    qu = None
+
+from pyzx.graph import Graph
+from pyzx.utils import EdgeType, VertexType
+from pyzx.mapping import to_tensor
+
+SEED = 1337
+
+@unittest.skipUnless(np, "numpy needs to be installed for this to run")
+@unittest.skipUnless(qu, "quimb needs to be installed for this to run")
+class TestMapping(unittest.TestCase):
+    def test_id_tensor(self):
+        g = Graph()
+        x = g.add_vertex(VertexType.BOUNDARY)
+        y = g.add_vertex(VertexType.BOUNDARY)
+        g.add_edge(g.edge(x, y), edgetype = EdgeType.SIMPLE)
+        
+        tn = to_tensor(g)
+        self.assertTrue((tn & qtn.Tensor(data = [0, 1], inds = ("0",)) 
+                            & qtn.Tensor(data = [0, 1], inds = ("1",)))
+                        .contract(output_inds = ()) == 1)
+        self.assertTrue((tn & qtn.Tensor(data = [1, 0], inds = ("0",)) 
+                            & qtn.Tensor(data = [1, 0], inds = ("1",)))
+                        .contract(output_inds = ()) == 1)
+    
+    def test_hadamard_tensor(self):
+        g = Graph()
+        x = g.add_vertex(VertexType.BOUNDARY)
+        y = g.add_vertex(VertexType.BOUNDARY)
+        g.add_edge(g.edge(x, y), edgetype = EdgeType.HADAMARD)
+        
+        tn = to_tensor(g)
+        self.assertTrue(abs((tn & qtn.Tensor(data = [1, 0], inds = ("0",)) 
+                            & qtn.Tensor(data = [1 / np.sqrt(2), 1 / np.sqrt(2)], inds = ("1",)))
+                        .contract(output_inds = ()) - 1) < 1e-9)
+        self.assertTrue(abs((tn & qtn.Tensor(data = [0, 1], inds = ("0",)) 
+                            & qtn.Tensor(data = [1 / np.sqrt(2), -1 / np.sqrt(2)], inds = ("1",)))
+                        .contract(output_inds = ()) - 1) < 1e-9)
+    
+    def test_xor_tensor(self):
```

Comment:
I would suggest adding a test (or changing this test) to include some phases on some of the spiders, to test whether you have implemented these phases correctly.